src/machine/machinectl.c

```diff
@@ -117,12 +118,63 @@ static int compare_machine_info(const void *a, const void *b) {
         return strcmp(x->name, y->name);
 }
 
+static void clean_machine_info(MachineInfo *machines, size_t n_machines) {
+        size_t i;
+
+        if (!machines || !n_machines)
+                return;
+
+        for (i = 0; i < n_machines; i++) {
+                free(machines[i].os);
+                machines[i].os = NULL;
+        }
+        free(machines);
+}
+
+static int get_os_name(sd_bus *bus, const char *name, char **out) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        const char *k, *v, *os;
+        int r;
+
+        assert(bus);
+        assert(name);
+        assert(out);
+
+        r = sd_bus_call_method(bus,
+                "org.freedesktop.machine1",
+                "/org/freedesktop/machine1",
+                "org.freedesktop.machine1.Manager",
+                "GetMachineOSRelease",
+                NULL, &reply, "s", name);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_enter_container(reply, 'a', "{ss}");
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        while ((r = sd_bus_message_read(reply, "{ss}", &k, &v)) > 0) {
+                if (streq(k, "PRETTY_NAME"))
+                        os = v;
+        }
+
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        r = sd_bus_message_exit_container(reply);
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        return free_and_strdup(out, os);
```

Comment:
Hmm, this is strange. Generally in our code if we have a function that returns something via a call-by-ref parameter, then we will never read from it, and only write to it, so that parameters are either input or output parameters, but not both (there are exceptions to this rule though). Hence I think free_and_strdup() in this case is not the right function to call. 

Moreover, in our sources functions are either supposed to all errors or no errors (in the latter case it's left to the caller to log about errors). Currently, you log about parse errors here, but not about memory errors. Please either log for all errors here (which appears to be the better option here, in my opinion), or for none. Also see CODING_STYLE about this.

My suggestion would hence be:

```c
c = strdup(os);
if (!c)
     return log_oom();

return 0;
```